frontend/public/module/k8s/pods.ts

```diff
@@ -177,26 +176,41 @@ export const podPhase = (pod): PodPhase => {
     return 'Terminating';
   }
 
-  let ret = pod.status.phase;
-  if (pod.status.reason) {
-    ret = pod.status.reason;
-  }
+  let initializing = false;
+  let phase = pod.status.phase || pod.status.reason;
 
-  if (pod.status.containerStatuses) {
-    pod.status.containerStatuses.forEach(function(container) {
-      if (container.state.waiting && container.state.waiting.reason) {
-        ret = container.state.waiting.reason;
-      } else if (container.state.terminated && container.state.terminated.reason) {
-        ret = container.state.terminated.reason;
+  if (pod.status.initContainerStatuses) {
+    pod.status.initContainerStatuses.forEach(function(container) {
```

Comment:
If you use `_.each`, you can skip the check for existence.

```js
_.each(pod.status.initContainerStatuses, container => {
```

Reply:
Yeah, I like that better